openedx/openedx-platform
# Update to django-storage-swift v1.2.18

Updates the SWIFT Django Storage requirement, to allow lazy SWIFT storage connections to be enabled by default.

Django takes great care to initialise Storage classes lazily, and so connecting on storage creation is generally a valid option. However there are some places in edx-platform where this is difficult to do, e.g., [the `CustomizableFileField` class](https://github.com/edx/edx-platform/blob/94f65768eed0c0a8c06cc8f512c6b2771af86470/openedx/core/djangoapps/content/block_structure/models.py#L78), which implicitly calls [`get_storage()`](https://github.com/edx/edx-platform/blob/94f65768eed0c0a8c06cc8f512c6b2771af86470/openedx/core/storage.py#L81-L91), which in turn instantiates a `Storage` instance.  Overall this results in a storage connections being made at module import time, which results in slow performance and unintended hits to SWIFT authentication.

**Discussions**:  https://github.com/dennisv/django-storage-swift/pull/97

**JIRA tickets**: [OSPR-2232](https://openedx.atlassian.net/browse/OSPR-2232)

**Sandbox URL**: TBD - sandbox is being provisioned. 

* LMS: https://pr17487.sandbox.opencraft.hosting/
* Studio: https://studio-pr17487.sandbox.opencraft.hosting/

**Merge deadline**: Please merge before Hawthorn is cut.

**Testing instructions**:

1. Set up swift storage on your devstack/server.
1. When starting edxapp or running basic `.manage.py` commands, notice that no SWIFT connections are made.
1. Upload a file to a discussion board post, and note that SWIFT connections are made during this process.  File should upload to the configured container as expected.

**Reviewers**
- [ ] edX reviewer[s] TBD -- CC @nedbat
